detail/graph.py

```diff
@@ -1,22 +1,21 @@
 """ Plots the workflow's graph
 """
 
-def get_aiida_link(cof_label, link_label):
+def get_aiida_link(group_node, extra_tag):
     import os
     from aiida.orm.querybuilder import QueryBuilder
-    from aiida.orm import  Node, StructureData, WorkFunctionNode
+    from aiida.orm import  Node, Group
 
     explore_url = os.getenv('EXPLORE_URL', "https://dev-www.materialscloud.org/explore/curated-cofs")
 
     qb = QueryBuilder()
-    qb.append(Node, filters={'label': cof_label}, tag='cof')
-    qb.append(WorkFunctionNode, filters={'attributes.function_name': {'==': 'link_outputs'}}, with_incoming='cof', tag='link')
-    qb.append(Node, edge_filters={'label': link_label}, with_outgoing='link')
+    qb.append(Group, filters={'uuid': group_node.uuid}, tag='group')
+    qb.append(Node, filters={'extras.curated-cof_tag': extra_tag}, with_group='group')
```

Comment:
This is still one query per node (of course, they will be already faster than before). 
How about you do the following in `main.ipynb`:

 1. load the group (you already do)
 2. 
```
node_dict = {}
for node in self.group.nodes:
   node_dict[node.get_extra('curated-cof_tag')] = node
self.node_dict = node_dict
```
 3. use the node_dict without having to query